osu.Game/Screens/Ranking/Expanded/Statistics/CounterStatistic.cs

```diff
@@ -19,7 +19,7 @@ public class CounterStatistic : StatisticDisplay
         private readonly int count;
         private readonly int? maxCount;
 
-        private RollingCounter<int> counter;
+        protected RollingCounter<int> Counter { get; private set; }
```

Comment:
this is weird abuse of this class. you're initialising a `CounterStatistic` with a readonly count and then updating the count at runtime. the count variable should probably not exist.

Reply:
How about making both count variables nullable and construct `PerformanceStatistic` with null counter values and hook into the counter directly? Unless making a carbon copy of `PerformanceStatistic` is fine, which I doubt ...

Reply:
I think `PerformanceStatistic` should inherit from `StatisticDisplay` instead of `CounterStatistic`, but also `CounterStatistic.StatisticCounter` (lol) - the inner rolling counter class - should be extracted out to avoid duplication. That was what I was mostly worried about, the inner class and only it.

Reply:
It took me 3 reads to get it correctly , but yeah on a second thought that approach seems cleaner than patching a class to control directly the counter. Will give it a shot soon ..